src/Database/LSMTree/Internal/MergingRun.hs

```diff
@@ -743,8 +752,9 @@ completeMerge mergeVar mergeKnownCompletedVar = do
       (OngoingMerge rs m) -> do
         -- first try to complete the merge before performing other side effects,
         -- in case the completion fails
-        --TODO: Run.fromMutable claims not to be exception safe
-        -- may need to use uninteruptible mask
+        --TODO: Run.fromMutable (used in Merge.complete) claims not to be
+        -- exception safe so we should probably be using the resource registry
+        -- and test for exception safety.
```

Comment:
Makes sense. We could use `modifyWithActionRegistry` or something similar to guarantee that the result of `Run.fromMutable` is put in the `mergeVar`. 

I'm hoping we can make `Run.fromMutable` and `Merge.complete` exception safe without passing a registry into it, so that we don't have to have a registry argument in too many places, but I haven't tried that yet